packages/consumption/src/modules/requests/itemProcessors/proposeAttribute/ProposeAttributeRequestItemProcessor.ts

```diff
@@ -120,7 +120,14 @@ export class ProposeAttributeRequestItemProcessor extends GenericRequestItemProc
 
             const latestSharedVersion = await this.consumptionController.attributes.getSharedVersionsOfAttribute(parsedParams.attributeId, [requestInfo.peer], true);
 
-            if (latestSharedVersion.length === 0) {
+            const wasSharedBefore = latestSharedVersion.length > 0;
+            const wasDeletedByPeerOrOwner =
+                latestSharedVersion[0]?.deletionInfo?.deletionStatus === DeletionStatus.DeletedByPeer ||
+                latestSharedVersion[0]?.deletionInfo?.deletionStatus === DeletionStatus.DeletedByOwner;
+            const isLatestSharedVersion = latestSharedVersion[0]?.shareInfo?.sourceAttribute?.toString() === existingSourceAttribute.id.toString();
+            const predecessorWasSharedBefore = wasSharedBefore && !isLatestSharedVersion;
```

Comment:
This hardly assumes that the method `canAccept` is called before the method `accept` is called and therefore no successor of the existing Attribute could have been shared, right?

Reply:
Exactly!

Reply:
Nice way to take advantage of that!

Reply:
But perhaps it would be nice if you also add a test regarding the error "You cannot share the predecessor of an already shared Attribute version." in canAccept.